C#/cs01/13.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Collections;
+namespace Brainchild.Curriculum.Solutions
+{
+    //Implement push and pop using Generics.
+    public class StackPushPop<T>
+    {
+        static int MAX = 1000;
+        static int index=-1;
+        T[] stack = new T[MAX];
+        public void Push(T data)
+        {
+            index = index + 1;
+            stack[index] = data;
+        }
+        public T Pop()
+        {
+            T[] stackNew=new T[MAX];
```

Comment:
why do you need a new array here?

Reply:
To delete the last element of the stack

Reply:
you dont have the delete anything. you can just increment  or decrement the counter ? counter is internal state which points to the top of the stack. in your case it is the index ?  techincally there is no need of printing al the elements of a stack.

Reply:
Printing the array can be done without deleting the last element. However we need to compare the elements in Assert when we compare the arrays. So that would require the last element to be deleted from the result array